src/rpc.rs

```diff
@@ -0,0 +1,56 @@
+use crate::abi;
+use substreams::{scalar::BigInt, Hex};
+use substreams_ethereum::rpc::RpcBatch;
+
+pub fn collection_data_call(collection_address: Vec<u8>) -> String {
+    let responses = RpcBatch::new()
+        .add(
+            abi::collections_v2::functions::Creator {},
+            collection_address,
+        )
+        .execute()
+        .unwrap()
+        .responses;
+
+    let creator: String =
+        match RpcBatch::decode::<_, abi::collections_v2::functions::Creator>(&responses[0]) {
+            Some(data) => Hex(data).to_string(),
+            None => {
+                panic!("Failed to decode fee growth global 1x128");
+            }
+        };
+
+    return creator;
+}
+
+pub fn get_collection_item_count(collection_address: Vec<u8>) -> Option<BigInt> {
+    substreams::log::info!("get_collection_item_count2 {:?}", collection_address);
+    let func = abi::collections_v2::functions::ItemsCount {};
+    if let Some(items_result) = func.call(collection_address) {
+        return Some(items_result);
+    };
+
+    return None;
```

Comment:
I think you can avoid the checks and directly return the result
```suggestion
    fun.call(collection_address)
```

Reply:
oohh true, updated!